src/datastore/postgres-store.ts

```diff
@@ -3602,7 +3682,7 @@ export class PgDataStore
       index_block_hash,
       atch_resolved,
     } = bnsName;
-    await client.query(`UPDATE names SET latest = $1 WHERE name= $2`, [false, name]);
+    if (canonical) await client.query(`UPDATE names SET latest = $1 WHERE name= $2`, [false, name]);
```

Comment:
@zone117x found this bug as well. For example, if we receive a non-canonical name, It was immediately being set out to latest false and canonical true.